src/Stack/Snapshot.hs

```diff
@@ -260,17 +262,29 @@ loadResolver (ResolverCustom url loc) = do
 
     load :: FilePath -> RIO env SnapshotDef
     load fp = do
+      let resolveLocalArchives sd = sd {
+            sdLocations = resolveLocalArchive <$> sdLocations sd
+          }
+          resolveLocalArchive (PLOther (PLArchive archive)) = 
+            PLOther $ PLArchive $ archive {
+              archiveUrl = T.pack $ resolveLocalFilePath (T.unpack $ archiveUrl archive)
+            }
+          resolveLocalArchive pl = pl
+          resolveLocalFilePath path =
+            if not $ isURI path && FilePath.isRelative path
```

Comment:
Hmm, I think this condition is saying "If it's not a uri and relative"  AFAICT that should never happen.

Perhaps the logic should be `if not (isURI path) && FilePath.isRelative path`.  Also I recommend trying it manually on your real world case to ensure that it is indeed a fix.

Reply:
Hmm, yeah I guess if `path` is absolute then the behavior of `</>` will forget about the path on the left.  Seems like perhaps the isRelative test is not necessary at all here?  Would be good to have a comment about it.

Reply:
Hmm, I am not sure the "forget about the left path" is very obvious although it is indeed documented like that.

I'd stick to "don't touch if it is either a URI or an absolute path".

I'll submit another commit later this week and create a PR to fix master. I won't be able to test my exact use case because of #3861 though.